File: Data Structures & Algorithms/buy-and-sell-crypto/submission-4.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int maxProfit = 0;
        int buy = prices[0];

        for (int num : prices) {
            maxProfit = max(maxProfit, num - buy);
            buy = min(buy, num);
        }

        return maxProfit;
    }
};

File: Data Structures & Algorithms/max-water-container/submission-4.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        l = 0
        r = len(heights)-1

        curMax = 0

        while l < r:
            curMax = max(curMax, min(heights[l], heights[r]) * (r-l))

            if heights[l] >= heights[r]:
                r -= 1
            else:
                l += 1

        return curMax